src/utils/formatCoffeeScriptAst.ts

```diff
@@ -70,7 +72,7 @@ function shouldTraverse(value) {
  * CoffeeScript AST nodes are always instances of a custom class, so use the
  * constructor name to distinguish between node children and non-node children.
  */
-function isNode(value) {
+function isNode(value: {}): boolean {
```

Comment:
Oh, yeah, you're right, `instanceof Base` works just fine. My line of reasoning was "it's not exported by CoffeeScript and types are erased, so there's no way TypeScript can generate code that works". But it actually is exported by CoffeeScript, just not at the top level. Will fix in a follow-up commit. (But for `shouldTraverse`, I think `any` is probably still reasonable.)

And thanks for pointing me to the `value is Node` syntax (user-defined type guards). I had no idea TypeScript had that.